### PR TITLE
Lesson plan PDF/zip format fixes

### LINKED ISSUE #478: File names in zip archive are URI escaped
```
$ unzip ~/Downloads/blustery-container-designed-mutable.zip
Archive:  ~/Downloads/blustery-container-designed-mutable.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
    35406  2018-05-03 10:23   01-Two-Factor%20Authentication%20-%20Beginning.pdf
    36019  2018-05-03 10:23   02-Locking%20Down%20Social%20Media%20-%20Beginning.pdf
---------                     -------
    71425                     2 files
```

This only the case when the zip file was made by nginx (the `ZipMiddleware` used for dev is incorrectly unescaping the file name, in other words).

### PR DESCRIPTION
**Pull request by wioux**
To inline assets into the PDF template we are using `Rails.application.assets`, but this value isn't loaded in production where we have `config.assets.compile = false`. This branch fixes the issue by looking for the precompiled file under public/assets.

It also fixes #478 by not URI escaping filenames in the nginx zip manifest.

### CODE REVIEW

**Comment by k-stewart** [comment]
Pls add tests on these two issues!

File: app/helpers/pdf_helper.rb [hunk 1]
```diff
@@ -3,15 +3,28 @@ def page_break
...
   def pdf_style_sheet
+    asset_data = get_asset("pdf.css")
     content_tag(:style, type: "text/css") do
-      Rails.application.assets.find_asset("pdf.scss").to_s.html_safe # rubocop:disable Rails/OutputSafety
+      asset_data.source.html_safe # rubocop:disable Rails/OutputSafety
+    end
+  end
+
+  def get_asset(asset)
+    if Rails.application.config.assets.compile
```
Comment (k-stewart) [inline app/helpers/pdf_helper.rb]: Does this compile all the assets manually each time this method is called?  That seems very time-consuming.

Comment (wioux) [reply app/helpers/pdf_helper.rb]: No it's just a boolean variable (indicating whether or not Sprockets is around to compile assets for you).

**Comment by wioux** [comment]
Thanks for the review! I added some tests.

File: spec/controllers/concerns/zipping_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  let(:controller) do
+    controller_class.new.tap do |c|
+      c.response = ActionDispatch::Response.new
+    end
+  end
+
+  let(:files) do
+    5.times.map do |i|
+      double(
```
Comment (k-stewart) [inline spec/controllers/concerns/zipping_spec.rb]: This double is beautiful

Comment (k-stewart) [reply spec/controllers/concerns/zipping_spec.rb]: It does exactly what it needs to, and no more.  so perfect!

**Comment by k-stewart** [comment]
These tests make me unreasonably happy.

**Pull request merged**